pkg/controller/hawtio/certificate.go

```diff
@@ -83,3 +86,64 @@ func generateCertificateSecret(name string, namespace string, caSecret *corev1.S
 		}, Type: corev1.SecretTypeTLS,
 	}, nil
 }
+
+func ValidateCertificate(caSecret corev1.Secret, validAtLeastForHours float64) bool {
+	block, _ := pem.Decode(caSecret.Data[corev1.TLSCertKey])
+	cert, err := x509.ParseCertificate(block.Bytes)
+
+	if err != nil {
+		log.Error(err, "certificate reading error")
+	}
+
+	diff := cert.NotAfter.Sub(time.Now()).Hours()
+	// if cert is valid longer than certain amount of hours
+	if diff > validAtLeastForHours {
+		log.Info(fmt.Sprintf("Certificate is valid for %.0f days", diff/24))
+		return true
+	}
+	//if is valid
+	return false
+}
+
+func createCertValidationCronJob(name, namespace, hawtioName, image string) *v1beta1.CronJob {
+	cronjob := &v1beta1.CronJob{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+		Spec: v1beta1.CronJobSpec{
+			Schedule:          "* */12 * * *",
```

Comment:
Maybe we could have this configurable in the `Hawtio` CRD, maybe in: https://github.com/hawtio/hawtio-operator/blob/d9c52667bf7bf1b76341d424d75c0976673b2c40/pkg/apis/hawtio/v1alpha1/hawtio_types.go#L81-L85

Also we could add a parameter to enable the automatic rotation.